src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexSearcherHolder.cs

```diff
@@ -65,43 +64,50 @@ internal IndexSearcherHoldingState GetStateHolder(Transaction tx)
                     continue;
                 }
 
-                Interlocked.Increment(ref state.Usage);
-
                 return state;
             }
 
             throw new InvalidOperationException($"Could not get an index searcher state holder for transaction {txId}");
         }
 
-        public void Cleanup(long oldestTx)
+
+        public void Cleanup(long oldestTx, bool everything = false)
```

Comment:
What's the meaning of `"everything"`? What does it mean - "cleanup but not everything". Can we have more meaningful name or maybe an enum here.

Reply:
Normal clean will keep one `IndexSearcherHoldingState` which might hold an `IndexSearcher`.
`everything` will clean even that last one.

I don't like the name as well, do have an idea for a better name?